megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -2157,6 +2170,172 @@ private void launchFighters(MovePath path) {
+     * manual fire if the game options allow it, with any change taking effect next round.
+     * Normal setting is to have the AMS active/automatic. It may be turned off to conserve
+     * ammo on relatively undamaged units, and laser AMS may be turned off to help reduce
+     * overheating. Manual use is reserved as an emergency anti-infantry measure.
+     *
+     */
+    private void setAMSModes() {
+
+        // Get conventional infantry if manual mode is available
+        List<Entity> enemyInfantry = new ArrayList<>();
+        if (game.getOptions().booleanOption(OptionsConstants.ADVCOMBAT_TACOPS_MANUAL_AMS)) {
+            for (Entity curEnemy : this.getEnemyEntities().stream().filter(Entity::isDeployed).collect(Collectors.toSet())) {
+                if (curEnemy.getPosition() != null && curEnemy.isVisibleToEnemy()) {
+
+                    if (curEnemy instanceof Infantry && !(curEnemy instanceof EjectedCrew)) {
+                        enemyInfantry.add(curEnemy);
+                    }
+
+                }
+            }
+        }
+
+        for (Entity curEntity : this.getEntitiesOwned()) {
+            if (!curEntity.isDeployed() || curEntity.getPosition() == null) {
+                continue;
+            }
+
+            List<WeaponMounted> activeAMS = curEntity.
+                    getWeaponList().
+                    stream().
+                    filter(w -> w.getType().hasFlag(AmmoWeapon.F_AMS) && w.hasModes()).
+                    collect(Collectors.toList());
+
+            if (!activeAMS.isEmpty()) {
+
+                // Set default to on/automatic and test to see if it should be off or manual instead
+                EquipmentMode newAMSMode = EquipmentMode.getMode(Weapon.MODE_AMS_ON);
+
+                boolean isOverheating = (curEntity instanceof Mech) && (curEntity.getHeat() >= 14);
+
+                // If there are enough nearby enemy infantry (only counted if the game option is
+                // set), choose manual fire
+                if (!enemyInfantry.isEmpty() && !curEntity.isAirborne()) {
+                    int infantryRange = enemyInfantry.stream().mapToInt(e -> Compute.effectiveDistance(game, curEntity, e)).min().getAsInt();
+                    if (infantryRange <= 3) {
+                        newAMSMode = EquipmentMode.getMode(Weapon.MODE_AMS_MANUAL);
+                    }
+                }
+
+                // If AMS was used manually this round, chances are it will be needed next round too
+                if (usedManualAMS(curEntity.getId())) {
+                    newAMSMode = EquipmentMode.getMode(Weapon.MODE_AMS_MANUAL);
+                }
+
+                for (WeaponMounted curAMS : activeAMS) {
+
+                    EquipmentMode curMode = curAMS.curMode();
+
+                    // Turn off laser AMS to help with overheating problems
+                    if (curAMS.getType().hasFlag(WeaponType.F_ENERGY)) {
+                        if (isOverheating) {
```

Comment:
As above.  Also, while we apply stricter overheating checks to ASF than to 'mechs in other sections of the Princess code, it looks like isOverheating would ignore ASFs, causing them to be less likely to disable their LAMS units.

Reply:
Deliberate choice - fighters are more vulnerable to damage so having AMS active, even when overheating, is considered useful.  I considered included them at first but decided just to do Mechs and keep the complexity down.  Can look into adding something if warranted, though.